lib/cuda/cudaRFIS012.cpp

```diff
@@ -191,8 +191,16 @@ cudaEvent_t cudaRFIS012::execute(cudaPipelineState& /*pipestate*/,
 
     rfi_S012.set_metadata(voltage.get_metadata());
     const auto& rfi_S012_meta = rfi_S012.get_metadata();
-    auto freq_upchan_factor = rfi_S012_meta->get_freq_upchan_factor();
-    auto time_downsampling_fpga = rfi_S012_meta->get_time_downsampling_fpga();
+    std::vector<int> freq_upchan_factor;
+    if (rfi_S012_meta->has_freq_upchan_factor())
```

Comment:
There should always be a frequency upchannelization factor. It would always be 1 for this kernel, but it needs to be there.